src/pages/tonies/EncoderPage.tsx

```diff
@@ -0,0 +1,210 @@
+  HiddenDesktop,
+  StyledBreadcrumb,
+  StyledContent,
+  StyledLayout,
+  StyledSider,
+} from "../../components/StyledComponents";
+import type { DragEndEvent } from "@dnd-kit/core";
+import { DndContext, PointerSensor, useSensor } from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import { Button, message, Upload, Typography, Flex, Collapse } from "antd";
+import type { UploadProps } from "antd";
+import { useState } from "react";
+import { ToniesSubNav } from "../../components/tonies/ToniesSubNav";
+import { FileBrowser } from "../../components/tonies/FileBrowser";
+import { DraggableUploadListItem } from "../../components/tonies/DraggableUploadListItem";
+import { MyUploadFile, upload } from "../../util/encoder";
+import { createQueryString, getFilePathFromQueryParam } from "../../util/url";
+
+const { Text, Paragraph } = Typography;
+
+export const EncoderPage = () => {
+  const { t } = useTranslation();
+  const location: Location = useLocation();
+
+  const [fileList, setFileList] = useState<MyUploadFile[]>([]);
+  const [uploading, setUploading] = useState(false);
+  const [tafFilename, setTafFilename] = useState("UnnamedTonieFile");
+
+  const sensor = useSensor(PointerSensor, {
+    activationConstraint: { distance: 10 },
+  });
+
+  const onDragEnd = ({ active, over }: DragEndEvent) => {
+    if (active.id !== over?.id) {
+      setFileList((prev) => {
+        const activeIndex = prev.findIndex((i) => i.uid === active.id);
+        const overIndex = prev.findIndex((i) => i.uid === over?.id);
+        return arrayMove(prev, activeIndex, overIndex);
+      });
+    }
+  };
+
+  const onChange: UploadProps["onChange"] = ({ fileList: newFileList }) => {
+    setFileList(newFileList as MyUploadFile[]);
+  };
+
+  const onRemove = (file: MyUploadFile) => {
+    const index = fileList.indexOf(file);
+    const newFileList = fileList.slice();
+    newFileList.splice(index, 1);
+    setFileList(newFileList);
+  };
+
+  const handleUpload = async () => {
+    setUploading(true);
+    const formData = new FormData();
+
+    for (const file of fileList) {
+      await new Promise((resolve, reject) =>
+        upload(resolve, reject, formData, fileList, file)
+      );
+    }
+
+    const currentUnixTime = Math.floor(Date.now() / 1000);
+    const queryParams = {
+      name: tafFilename,
+      audioId: currentUnixTime - 0x50000000,
+      path: getFilePathFromQueryParam(location),
+      special: "library",
+    };
+
+    const queryString = createQueryString(queryParams);
+    const response = await fetch(
+      `${process.env.REACT_APP_TEDDYCLOUD_API_URL}/api/pcmUpload?${queryString}`,
+      {
+        method: "POST",
+        body: formData,
+      }
+    );
+
+    const responseData = await response.text();
+    if (response.ok) {
+      message.success(t("tonies.encoder.uploadSuccessful"));
+      setFileList([]);
+      setTafFilename("UnnamedTonieFile");
```

Comment:
Please change the filename for example to "NewContent.taf" to add the ".taf" file ending.